File: Python3/프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    #카펫의 노란색과 갈색으로 색칠된 격자의 개수는 기억했으나, 전체 카펫의 크기는 기억하지 못함
    #카펫의 가로, 세로 크기를 순서대로 배열에 담기
    
    for i in range(1, yellow+1):
        if yellow % i != 0:
            continue
            
        j = yellow // i
        if (i+2) * (j+2) - i*j == brown:
            return [j+2, i+2]
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    # for i in range(1, yellow+1):
    #     if yellow % i == 0:
    #         j = yellow // i
    #         if i >= j and yellow + brown == (i+2) * (j+2):
    #             return [i+2, j+2]

File: C++/프로그래머스/2/42842. 카펫/카펫.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int brown, int yellow) {
    vector<int> answer;
    for (int i = 1; i < yellow + 1; i++) {
        if (yellow % i != 0) {
            continue;
        }
        
        int j = yellow / i;
        if ((j+2) * (i+2) == yellow + brown) {
            answer.push_back(j+2);
            answer.push_back(i+2);
            return answer;
        }
    }
    return answer;
}